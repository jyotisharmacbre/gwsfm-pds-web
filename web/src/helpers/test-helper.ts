import checkPropTypes from 'check-prop-types';

export const findByTestAtrr = (component: any, attr: string) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component: any, expectedProps: any) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    'props',
    component.name
  );
  return propsErr;
};
