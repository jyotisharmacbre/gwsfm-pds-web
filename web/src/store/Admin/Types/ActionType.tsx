export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
