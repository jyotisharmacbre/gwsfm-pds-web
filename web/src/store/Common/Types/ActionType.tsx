export enum ActionType {
  DEFAULT = 'DEFAULT'
}
