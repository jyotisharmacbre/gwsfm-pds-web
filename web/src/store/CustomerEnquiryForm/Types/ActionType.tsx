export enum ActionType {
  DEFAULT = 'DEFAULT',
  PROJECT_ADD = 'PROJECT_ADD',
  PROJECT_ADD_ERROR = 'PROJECT_ADD_ERROR',
  PROJECT_EDIT_SUCCESS = 'PROJECT_EDIT_SUCCESS',
  GET_ENQUIRY_OVERVIEW_SUCCESS = 'GET_ENQUIRY_OVERVIEW_SUCCESS',
  GET_ENQUIRY_OVERVIEW_ERROR = 'GET_ENQUIRY_OVERVIEW_ERROR',
  GET_PROJECT_DETAIL_SUCCESS = 'GET_PROJECT_DETAIL_SUCCESS',
  GET_PROJECT_DETAIL_ERROR = 'GET_PROJECT_DETAIL_ERROR',
  RESET_PROJECT_DETAIL_STATE = 'RESET_PROJECT_DETAIL_STATE',
  SET_PROJECT_ID_STATE = 'SET_PROJECT_ID_STATE'
}
