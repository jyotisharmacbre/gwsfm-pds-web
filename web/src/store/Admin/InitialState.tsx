import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
