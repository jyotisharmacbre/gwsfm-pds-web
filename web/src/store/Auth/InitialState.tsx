import { IAuth } from "./Types/IAuth";

export const initialState: IAuth= {
    token: null,
    name: null,
    firstName: null,
    email: null
};
