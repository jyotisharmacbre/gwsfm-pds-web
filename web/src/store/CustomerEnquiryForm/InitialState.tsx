import { IProjectDetailState } from './Types/IProjectDetailState';
import EventType from '../../enums/EventType';
import Notify from '../../enums/Notify';

export const initialState: IProjectDetailState = {
    form: {
        projectId: '',
        name: '',
        contractorId: '',
        companyId: '',
        headOfProject: '',
        projectOwner: '',
        projectManager: '',
        pmHasExperience: false,
        scope: '',
        cnNumber: '',
        status: 1,
        engagementId: undefined,
        countryId: 0,
        currencyId: 0,
        probabilityOfWinning: '',
        approxValue: '',
        contractTypeId: '',
        cdmNotifiable: false,
        firstAssetWorkedOn: undefined,
        secondAssetWorkedOn: undefined,
        thirdAssetWorkedOn: undefined,
        comment: '',
        otherCompanyName: '',
        otherContractName: '',
        divisionId: '',
        otherDivision: '',
        businessUnitId: '',
        otherBusinessUnit: '',
        weightedTCV: undefined,
        soldMargin: undefined,
        otherEngagementType: '',
        otherFirstAssetWorkedOn: '',
        otherSecondAssetWorkedOn: '',
        otherThirdAssetWorkedOn: ''
    },
    enquiryOverview: {
        projectName: '',
        contractorId: '',
        headOfProject: '',
        projectManager: '',
        scope: '',
        cnNumber: '',
        otherContractName: ''
    },
    error: null,
    loading: false,
    notify: Notify.none,
    event: EventType.none,
    enquiryOverviewError: null
};