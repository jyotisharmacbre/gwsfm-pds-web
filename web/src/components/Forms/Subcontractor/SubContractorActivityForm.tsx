import React from 'react';
import { Field, FieldArray, reduxForm, InjectedFormProps, FormSection, formValueSelector } from 'redux-form';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import PdsFormSelect from '../../PdsFormHandlers/PdsFormSelect';
import PdsFormTextArea from '../../PdsFormHandlers/PdsFormTextArea';
import PdsFormButton from '../../PdsFormHandlers/PdsFormButton';
import PdsFormTypeAhead from '../../PdsFormHandlers/PdsFormTypeAhead';
import Quote from './Quote';
import { ISubContractorActivity } from '../../../store/SubContractor/Types/ISubContractorActivity';
import IReactIntl from '../../../Translations/IReactIntl';
import { FormattedMessage, injectIntl } from 'react-intl';
import FontawsomeSvg from '@fortawesome/fontawesome-svg-core';
import FontawsomeFree from '@fortawesome/free-solid-svg-icons';
import FontawsomeReact, { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTrash } from '@fortawesome/free-solid-svg-icons';
import { Validate, alphaNumeric, onlyNumber } from '../../../helpers/fieldValidations';
import { selectionButtons } from '../../../helpers/constants';
import { faPlusCircle } from '@fortawesome/free-solid-svg-icons';
import { newActivity } from '../../../store/SubContractor/InitialState';
import { connect } from 'react-redux';
import { IState } from '../../../store/state';
import { calculateSell } from '../../../helpers/formulas';

import NewTypeAhead from '../../TypeAhead/NewTypeAhead';
import { confirmAlert } from '../../Popup/CustomModalPopup';

import ValidatedNumericInput from '../../../components/NumericInput/index';
import { restrictMinus, restrictMinusAndAllowDecimal, restrictMinusAndAllowDecimalForMaxRangeHundred } from '../../../helpers/utility-helper';
interface Props {
	fields: any;
	activities: Array<ISubContractorActivity>;
	currencySymbol: string;
  intl: any;
  getListOfSubContractorData: (value: any) => Promise<any>; 
}

const SubContractorActivityForm: React.FC<Props> = (props: Props) => {
	const { fields, intl } = props;

	const formatSubcontractForTypeAhead = (data) => {
		let result: any = [];
		data.map((subcontractData: any) => {
			result.push({ label: subcontractData.name, id: subcontractData.subContractorId });
		});
		return result;
	};

	return (
		<div>
			{fields.map((member, index) => (
				<div className="row" key={index} data-test="sub-contractor-form">
					<div className="col-lg-12">
						<div className="forms_wrap">
							{fields.length > 1 ? (
								<button
									data-test="deleteactivity"
									className="delete_text"
									onClick={() =>
										confirmAlert({
											intl: props.intl,
											titleKey: 'TITLE_CONFIRMATION',
											contentKey: 'MESSAGE_SUBCONTRACTOR_DELETE_ACTIVITY',
											handleConfirm: () => fields.remove(index)
										})}
								>
									<FormattedMessage id="BUTTON_DELETE" />
									<FontAwesomeIcon className="" icon={faTrash} />
								</button>
							) : null}
							<div className="row">
								<div className="col-lg-7">
									<Field
										name={`${member}.activityName`}
										data-test="activityName"
										type="text"
										component={PdsFormInput}
										validate={[ Validate.maxLength(250) ]}
										labelKey="LABEL_ACTIVITY_NAME"
										placeholderKey="PLACEHOLDER_ACTIVITY_NAME"
									/>
									<Field
										name={`${member}.isExistingSubcontractor`}
										component={PdsFormButton}
										buttons={selectionButtons}
										labelKey="LABEL_EXISTING_SUBCONTRACTOR"
									/>
									<NewTypeAhead
										name={`${member}.subcontractorId`}
										onSearch={props.getListOfSubContractorData}
										formatData={formatSubcontractForTypeAhead}
										DynamicsType="subcontractorId"
										placeholderKey="PLACEHOLDER_SUBCONTRACTOR"
										labelName="LABEL_SUBCONTRACTOR"
										validationKey="LABEL_SUBCONTRACTOR"
										submitParam="id"
										className="required"
									/>

									{props.activities[index].subcontractorId === '0' && (
										<Field
											name={`${member}.otherSubcontractorName`}
											type="text"
											component={PdsFormInput}
											labelKey="LABEL_OTHER_SUBCONTRACTOR"
											placeholderKey="PLACEHOLDER_OTHER_SUBCONTRACTOR"
											className="required"
											validate={[Validate.required('LABEL_OTHER_SUBCONTRACTOR'), Validate.maxLength(250)]}
										/>
									)}

									<Field
										name={`${member}.isPreferredSupplier`}
										component={PdsFormButton}
										buttons={selectionButtons}
										labelKey="LABEL_PREFERRED_SUPPLIER"
									/>
									<Field
										name={`${member}.totalCost`}
										type="text"
										component={PdsFormInput}
										className="width-120 pl-20"
										validate={[ Validate.maxLength(1000), onlyNumber ]}
										currency={props.currencySymbol}
										normalize={restrictMinusAndAllowDecimal}
										divPosition="relative"
										labelKey="LABEL_TOTAL_COST"
										placeholderKey=""
									/>
									<Field
										name={`${member}.grossMargin`}
										type="text"
										component={PdsFormInput}
										className="width-120 pl-20"
										validate={[ Validate.maxLength(1000), onlyNumber ]}
										currency="%"
										normalize={restrictMinusAndAllowDecimalForMaxRangeHundred}
										divPosition="relative"
										labelKey="LABEL_GROSS_MARGIN"
										placeholderKey=""
									/>
									<Field
										name={`${member}.totalSell`}
										type="number"
										input={{
											value: calculateSell(
												props.activities[index].totalCost,
												props.activities[index].grossMargin
											),
											disabled: true
										}}
										component={PdsFormInput}
										className="width-120 pl-20"
										currency={props.currencySymbol}
										divPosition="relative"
										labelKey="LABEL_TOTAL_SELL"
										placeholderKey=""
									/>
									<Field
										labelKey="LABEL_COMMENTS"
										name={`${member}.comments`}
										rows="7"
										component={PdsFormTextArea}
										validate={[ Validate.maxLength(5000) ]}
										placeholderKey="PLACEHOLDER_ADDITIONAL_COMMENTS"
									/>
								</div>
							</div>
							<FieldArray name={`${member}.quotes`} currencySymbol = {props.currencySymbol} component={Quote} key={index} />
						</div>
					</div>
				</div>
			))}
			<div className="newActiv_btn">
				<button
					data-test="addActivity"
					name="addActivity"
					type="button"
					disabled={fields.length > 4}
					className="active"
					onClick={() => fields.push({ ...newActivity })}
				>
					<FontAwesomeIcon className="" icon={faPlusCircle} />
					<FormattedMessage id="BUTTON_NEW_ACTIVITY" />
				</button>
			</div>
		</div>
	);
};

const mapStateToProps = (state: IState) => ({
	activities: selector(state, 'activities')
});

const selector = formValueSelector('subContractorForm');

export default connect(mapStateToProps)(SubContractorActivityForm);
