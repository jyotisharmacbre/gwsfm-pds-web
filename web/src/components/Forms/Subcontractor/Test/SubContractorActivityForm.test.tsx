import React from 'react';
import { mount, shallow } from 'enzyme';
import SubContractorActivityForm from '../SubContractorActivityForm';
import {initialState,newActivity} from '../../../../store/SubContractor/InitialState';
import {ActionType} from '../../../../store/SubContractor/Types/ActionType';
import subContractorReducer from '../../../../store/SubContractor/Reducer';
import {findByTestAtrr} from '../../../../helpers/test-helper';
import configureStore from 'redux-mock-store';
import { reducer as formReducer,reduxForm } from 'redux-form';
import { Provider } from 'react-redux';
 import { IntlProvider } from 'react-intl';
 import translations from '../../../../Translations/translation';
import renderer from "react-test-renderer";

describe('Sub Contractor Activity Form tests', () => {
  const subContractorFormAddAction: any = {
          type: ActionType.SUB_CONTRACTOR_ADD_NEW_ACTIVITY
  };
  let wrapper: any;
  const mockStore = configureStore([]);
  let store;
  const Decorated = reduxForm({ form: "subContractorForm" })(SubContractorActivityForm);
  beforeEach(() => {
    store = mockStore({
      form: {
        subContractorForm: {
          values: {
            activities: initialState.form.activities
          }
        }
      }
    }
    );
    wrapper = mount(
      <Provider store={store}>
        <IntlProvider locale="en" messages={translations['en'].messages}>
          <Decorated fields={initialState.form.activities}/>
        </IntlProvider>
      </Provider>
    );
  });
  
  it('Defines the component', () => {
    expect(wrapper).toBeDefined();
  }); 
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }); 
  it('renders the SubContractorActivityForm component with no errors', () => {
    expect(findByTestAtrr(wrapper,'sub-contractor-form').length).toEqual(1);
  });

  it('should not renders the delete button', () => {
    expect(findByTestAtrr(wrapper,'deleteactivity').length).toEqual(0);
  });

  it('test the add activity click event', () => {
    let addActivity = findByTestAtrr(wrapper,'addActivity');
    addActivity.simulate('click');
    wrapper.update();
   });
});
