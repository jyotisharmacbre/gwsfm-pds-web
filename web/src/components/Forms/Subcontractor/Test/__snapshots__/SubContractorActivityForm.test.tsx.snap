// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub Contractor Activity Form tests should match the snapshot 1`] = `ReactWrapper {}`;
